Source/Core/ProofGenInfoManager.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization.Formatters.Binary;
+
+namespace Microsoft.Boogie
+{
+  public static class ProofGenInfoManager
+  {
+    private static IDictionary<Implementation, ProofGenInfo> mapFromImplementationToProofGenInfo;
```

Comment:
I would rename `mapFromImplementationToProofGenInfo` to `implToProofGenInfo` to be more consistent with naming in other parts of the proof generation code. Rename the Getter Method accordingly.